torch/distributed/_functional_collectives.py

```diff
@@ -141,6 +141,18 @@ def broadcast(self: torch.Tensor, src: int, group: RANK_TYPES, tag: str = ""):
     tensor = torch.ops.c10d_functional.broadcast(self, src, tag, rankset, group_size)
     return _maybe_wrap_tensor(tensor)
 
+def scatter(self: torch.Tensor, scatter_list: List[torch.Tensor], src: int, group: RANK_TYPES, tag: str = ""):
```

Comment:
I think this looks wrong, the scatter API should be sth like this:

scatter(self: torch.Tensor, src: int, scatter_dim: int, group, ..) -> torch.Tensor

We don't need the scatter_list thing here, we only need to allocate them inside this torch.ops.c10d_functional.scatter implementation:
i.e. 
```
scatter_list = [torch.empty_like(shard) for shard in self.chunk(num_ranks)]
```
and only return one shard after scatter. `scatter` looks similar to `reduce_scatter_tensor`, where it takes a big tensor on all ranks but return a small tensor (the scattered shard, but it does not do reduction). 

We should also name this as `scatter_tensor` to distinguish with the c10d version